js/src/dropdown.js

```diff
@@ -430,14 +431,13 @@ class Dropdown extends BaseComponent {
           .forEach(elem => EventHandler.off(elem, 'mouseover', null, noop()))
```

Comment:
maybe you can also move this

``` js
     // If this is a touch-enabled device we remove the extra
      // empty mouseover listeners we added for iOS support
      if ('ontouchstart' in document.documentElement) {
        [].concat(...document.body.children)
          .forEach(elem => EventHandler.off(elem, 'mouseover', null, noop()))
      }
```
     
 on hide method, remove lines 433 to 442 and lines 405 to 407 and call:

 ```js
      
      context.hide()
 ```
      
      
I am only missing how you can pass the following information

```js

     if (event && event.type === 'click') {
        relatedTarget.clickEvent = event
      }
```

Reply:
Yes, I have these changes in another branch (I will move the changes for removing the event handler in this PR though)

Reply:
I'd like to mention that this doesn't work anyway.. `noop()` returns a new function everytime it is called. So `.off()` can't remove those handlers because `noop() != noop()`.

Reply:
Ah, true. I believe that @XhmikosR discussed this thing before and has a branch with these changes for whole codebase.

Reply:
Oh I didn't know that. I wanted to mention it since you're already on working on that code. 😅